src/commands/deployments/connectToGitHub.ts

```diff
@@ -19,7 +20,14 @@ export async function connectToGitHub(context: IActionContext, target?: GenericT
         node = <DeploymentsTreeItem>target.parent;
     }
 
-    await editScmType(node.root.client, node, context, ScmType.GitHub);
+    if (node instanceof WebAppTreeItem) {
+        node.deploymentsNode = new DeploymentsTreeItem(node.parent, node.client, await node.client.getSiteConfig(), await node.client.getSourceControl());
```

Comment:
You shouldn't be reassigning the deploymentsNode here.  You can either change the `SiteTreeItem` class to have an async create method like so: https://github.com/microsoft/vscode-azurefunctions/blob/master/src/tree/remoteProject/RemoteFunctionTreeItem.ts#L24-L31  This will ensure that the `deploymentsNode` is already defined.

The other thing you can do is check to see if `node.deploymentsNode` is undefined.  If it is, you can just refresh the treeItem and then proceed as normal.

Also, you should be checking if it's a `SiteTreeItem` rather than `WebAppTreeItem`, which extends `SiteTreeItem`.  This command can also be used on `SlotTreeItems` (which also extends `SiteTreeItem`)